### PR TITLE
Pass imagePullSecrets values to operator components via helm chart

### PR DESCRIPTION
**Pull request by e0ne**

### CODE REVIEW

File: deployment/sriov-network-operator/values.yaml [hunk 1]
```diff
@@ -35,3 +35,5 @@ images:
   sriovDevicePlugin: ghcr.io/k8snetworkplumbingwg/sriov-network-device-plugin
   resourcesInjector: ghcr.io/k8snetworkplumbingwg/network-resources-injector
   webhook: ghcr.io/k8snetworkplumbingwg/sriov-network-operator-webhook
+
+imagePullSecrets: []
```
Comment (SchSeba) [inline deployment/sriov-network-operator/values.yaml:39]: can you please add a new line here :)

**Review by SchSeba — COMMENTED** [review]
just a small nit looks good from my side

File: deployment/sriov-network-operator/README.md [hunk 2]
```diff
@@ -51,6 +51,10 @@ $ kubectl -n sriov-network-operator get pods
 In order to tailor the deployment of the network operator to your cluster needs
 We have introduced the following Chart parameters.
 
+| Name | Type | Default | description |
+| ---- |------|---------|-------------|
+| `imagePullSecrets` | list | `[]` | An optional list of references to secrets to use for pulling any of the SR-IOV Network Operator image if it's not overrided |
```
Comment (adrianchiris) [inline deployment/sriov-network-operator/README.md:56]: a laymans question : what may override this ?

File: bindata/manifests/operator-webhook/server.yaml [hunk 3]
```diff
@@ -47,6 +47,12 @@ spec:
       - key: "node.kubernetes.io/not-ready"
         operator: Exists
         effect: NoSchedule
+      {{- if .ImagePullSecrets }}
```
Comment (adrianchiris) [inline bindata/manifests/operator-webhook/server.yaml:50]: should we add also to NRI ? (bindata/manifests/webhook/)

**Review by adrianchiris — COMMENTED** [review]
so this PR actually does two things:

1. adds support for image pull secrets for the operator and its operands(sub components)
2. exposes a way to set image pull secrets via helm

It would be great if you could split to 2 commits and adjust PR description

File: deployment/sriov-network-operator/README.md (hunk 2 above)
Comment (e0ne) [reply deployment/sriov-network-operator/README.md:56]: fixed

File: deployment/sriov-network-operator/values.yaml (hunk 1 above)
Comment (e0ne) [reply deployment/sriov-network-operator/values.yaml:39]: done

File: bindata/manifests/operator-webhook/server.yaml (hunk 3 above)
Comment (e0ne) [reply bindata/manifests/operator-webhook/server.yaml:50]: done

**Review by adrianchiris — APPROVED** [review]

**Comment by adrianchiris** [comment]
@SchSeba can we merge this one ?

**Review by SchSeba — APPROVED** [review]
/lgtm

**Pull request merged**